skyportal/handlers/api/shift.py

```diff
@@ -538,24 +551,24 @@ def delete(self, shift_id, user_id):
         except ValueError:
             return self.error("Invalid user_id; unable to parse to integer")
 
-        su = (
-            ShiftUser.query_records_accessible_by(self.current_user, mode='delete')
-            .filter(ShiftUser.shift_id == shift_id)
-            .filter(ShiftUser.user_id == user_id)
-            .first()
-        )
-        if su is None:
-            return self.error(
-                "ShiftUser does not exist, or you don't have the right to delete him.",
-                status=403,
-            )
+        with self.Session() as session:
+            su = session.scalars(
+                ShiftUser.select(session.user_or_token, mode='delete')
+                .where(ShiftUser.shift_id == shift_id)
+                .where(ShiftUser.user_id == user_id)
+            ).first()
+            if su is None:
+                return self.error(
+                    "ShiftUser does not exist, or you don't have the right to delete him.",
+                    status=403,
+                )
 
-        DBSession().delete(su)
-        self.verify_and_commit()
-        self.push_all(
-            action='skyportal/REFRESH_SHIFTS', payload={'shift_id': int(shift_id)}
-        )
-        return self.success()
+            session.delete(su)
+            session.commit()
+            self.push_all(
+                action='skyportal/REFRESH_SHIFTS', payload={'shift_id': int(shift_id)}
```

Comment:
Maybe worth casting in a try catch at the beginning of the function?

Reply:
good idea.